src/_pytest/config/argparsing.py

```diff
@@ -362,6 +362,11 @@ def addoption(self, *optnames: str, **attrs: Any) -> None:
         results in help showing ``--two-words`` only, but ``--twowords`` gets
         accepted **and** the automatic destination is in ``args.twowords``.
         """
+        if getattr(self.parser, "after_preparse", False) and "default" not in attrs:
```

Comment:
this is not quite correct - the store_true action has a implied default, we should figure why its not applied as that details breaks argparse in some ways